repositories/riotrepository.go

```diff
@@ -0,0 +1,97 @@
+package repositories
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"riot-developer-proxy/internal/domain/entities"
+)
+
+type RGAPIRepository struct {
+	rgapiClient *rgapiClient
+}
+
+func NewRGAPIRepository(rgapiClient *rgapiClient) *RGAPIRepository {
+	return &RGAPIRepository{
+		rgapiClient: rgapiClient,
+	}
+}
+
+func (rr *RGAPIRepository) WithRegion(region string) {
+	rr.rgapiClient.WithRegion(region)
+}
+
+type SummonerDTO struct {
+	ID            string `json:"id"`
+	PUUID         string `json:"puuid"`
+	AccountID     string `json:"accountId"`
+	Name          string `json:"name"`
+	ProfileIconID int    `json:"profileIconId"`
+	Level         int    `json:"summonerLevel"`
+	RevisionDate  int64  `json:"revisionDate"`
+}
+
+func (rr *RGAPIRepository) GetAccountBySummonerName(ctx context.Context, summonerName string) (*SummonerDTO, error) {
+	resp, err := rr.rgapiClient.DoReq(ctx, "GET", "/lol/summoner/v4/summoners/by-name/"+url.PathEscape(summonerName))
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	if resp.StatusCode != http.StatusOK {
+		return nil, fmt.Errorf("err when getting account by summoner name Status: %d Response: %s", resp.StatusCode, string(body))
+	}
+
+	var summonerDTO SummonerDTO
+	if err := json.Unmarshal(body, &summonerDTO); err != nil {
+		return nil, err
+	}
+
+	return &summonerDTO, nil
+}
+
+type LeagueEntryDTO struct {
```

Comment:
nit: might be more readable to put all these structs at the top of the file

Reply:
Apply this suggestion improved the readability! Thanks